x/subscription/keeper/subscription.go

```diff
@@ -315,7 +308,20 @@ func (k Keeper) advanceMonth(ctx sdk.Context, subkey []byte) {
 		sub.DurationTotal += 1
 		k.resetSubscriptionDetailsAndAppendEntry(ctx, &sub, block, date)
 	} else {
-		if sub.AutoRenewal {
+		if sub.FutureSubscription != nil {
+			// Consumer made advance purchase. Now we activate it.
+			newSubInfo := sub.FutureSubscription
+
+			sub.Creator = newSubInfo.Creator
+			sub.PlanIndex = newSubInfo.PlanIndex
+			sub.PlanBlock = newSubInfo.PlanBlock
+			sub.DurationBought = newSubInfo.DurationBought
+			sub.DurationLeft = newSubInfo.DurationBought
+			sub.DurationTotal += 1
```

Comment:
reset duration total and update cluster

Reply:
Fixed in 87218cde444d5c5e591221242a5c0bc13aa83abb

Cluster is already updated in the function `resetSubscriptionDetailsAndAppendEntry`